Source/WebCore/animation/DeclarativeAnimation.h

```diff
@@ -76,13 +76,17 @@ class DeclarativeAnimation : public WebAnimation {
 
     void initialize(const RenderStyle* oldStyle, const RenderStyle& newStyle, const Style::ResolutionContext&);
     virtual void syncPropertiesWithBackingAnimation();
-    virtual Ref<DeclarativeAnimationEvent> createEvent(const AtomString& eventType, double elapsedTime, PseudoId) = 0;
+    virtual Ref<DeclarativeAnimationEvent> createEvent(const AtomString& eventType, std::optional<Seconds> scheduledTime, double elapsedTime, PseudoId) = 0;
     void invalidateDOMEvents(Seconds elapsedTime = 0_s);
 
 private:
     void disassociateFromOwningElement();
     AnimationEffectPhase phaseWithoutEffect() const;
-    void enqueueDOMEvent(const AtomString&, Seconds);
+    void enqueueDOMEvent(const AtomString&, Seconds elapsedTime, Seconds scheduledEffectTime);
```

Comment:
It's not clear to me what `scheduledEffectTime` is. Does it represent the start of the animation? Can you choose a better name?

Reply:
This is a term from the Web Animations specification, see https://www.w3.org/TR/web-animations/#scheduled-event-time.